src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventNotificationEmailManager.java

```diff
@@ -100,8 +97,7 @@ public void sendBookingStatusFilteredEmailForEvent(final IsaacEventPageDTO event
                 .put("event", event)
                 .build(),
             EmailType.SYSTEM);
-        log.debug(String.format("Sent email to user: %s %s, at: %s", user.getGivenName(), user.getFamilyName(),
-            user.getEmail()));
+        log.debug("Sent email to user: {} {}, at: {}", user.getGivenName(), user.getFamilyName(), user.getEmail());
```

Comment:
Huh. Not seen this before one of your recent tickets. Is this method preferred over string format for logging then?

Reply:
Yes, if you check SonarQube (or SonarLint in your IDE), it gives you a very detailed explanation about why the way we're using our log functions are not ideal. This is part of the explanation, but I'd recommend you to read the whole thing:

> Passing concatenated strings into a logging method can also incur a needless performance hit because the concatenation will be performed every time the method is called, whether or not the log level is low enough to show the message